tests/utils/test_alignment_utils.py

```diff
@@ -257,6 +257,91 @@ def add_dummy_measurements_to_track(track: SfmTrack) -> SfmTrack:
         assert np.allclose(aligned_sfm_result.get_track(0).point3(), np.array([1.0, 1.0, 0.0]))
         assert np.allclose(aligned_sfm_result.get_track(1).point3(), np.array([3.0, 3.0, 0.0]))
 
+    def test_ransac_align_poses_sim3_ignore_missing_pureidentity(self) -> None:
+        """Ensure that for identity poses, and thus identity Similarity(3), we get back exactly what we started with."""
+
+        aTi_list = [
+            Pose3(
+                Rot3(np.array([[0.771176, -0.636622, 0], [0.636622, 0.771176, 0], [0, 0, 1]])),
+                t=np.array([6.94918, 2.4749, 0]),
+            ),
+            Pose3(
+                Rot3(
+                    np.array([[0.124104, -0.992269, 0], [0.992269, 0.124104, 0], [0, 0, 1]]),
+                ),
+                t=np.array([6.06848, 4.57841, 0]),
+            ),
+            Pose3(
+                Rot3(
+                    np.array([[0.914145, 0.405387, 0], [-0.405387, 0.914145, 0], [0, 0, 1]]),
+                ),
+                t=np.array([6.47869, 5.29594, 0]),
+            ),
+            Pose3(
+                Rot3(np.array([[0.105365, -0.994434, 0], [0.994434, 0.105365, 0], [0, 0, 1]])),
+                t=np.array([5.59441, 5.22469, 0]),
+            ),
+            Pose3(
+                Rot3(np.array([[-0.991652, -0.12894, 0], [0.12894, -0.991652, 0], [0, 0, 1]])),
+                t=np.array([7.21399, 5.41445, 0]),
+            ),
+        ]
+        # Make twice as long.
+        aTi_list = aTi_list + aTi_list
+
+        bTi_list = copy.deepcopy(aTi_list)
+
+        aligned_bTi_list_est, aSb = alignment_utils.align_poses_sim3_ignore_missing(aTi_list, bTi_list)
+
+        for aTi, aTi_ in zip(aTi_list, aligned_bTi_list_est):
+            assert np.allclose(aTi.rotation().matrix(), aTi_.rotation().matrix(), atol=1e-3)
+            assert np.allclose(aTi.translation(), aTi_.translation(), atol=1e-3)
+
+
+    def test_ransac_align_poses_sim3_ignore_two_missing_estimated_poses(self) -> None:
```

Comment:
We should also cover the case where the missing cameras are different between the two inputs.

Reply:
That's a good idea, let's add in a follow-up.